tests/catch/unit/memory/hipArrayCreate.cc

```diff
@@ -129,20 +87,319 @@ TEST_CASE("Unit_hipArrayCreate_MultiThread") {
 
   devCnt = HipTest::getDeviceCount();
 
-  size_t tot, avail, ptot, pavail;
-  HIP_CHECK(hipMemGetInfo(&pavail, &ptot));
+  const size_t pavail = getFreeMem();
   for (int i = 0; i < devCnt; i++) {
-    threadlist.push_back(std::thread(ArrayCreateThreadFunc, i));
+    // TODO the HIP_CHECK and HIPASSERT are not threadsafe so this test is broken.
+    threadlist.push_back(std::thread(ArrayCreate_DiffSizes, i));
   }
 
-  for (auto &t : threadlist) {
+  for (auto& t : threadlist) {
     t.join();
   }
-  HIP_CHECK(hipMemGetInfo(&avail, &tot));
+  const size_t avail = getFreeMem();
 
   if (pavail != avail) {
     WARN("Memory leak of hipMalloc3D API in multithreaded scenario");
     REQUIRE(false);
   }
 }
 
+
+// All the possible formats for channel data in an array.
+static const std::vector<hipArray_Format> formats{
+    HIP_AD_FORMAT_UNSIGNED_INT8, HIP_AD_FORMAT_UNSIGNED_INT16, HIP_AD_FORMAT_UNSIGNED_INT32,
+    HIP_AD_FORMAT_SIGNED_INT8,   HIP_AD_FORMAT_SIGNED_INT16,   HIP_AD_FORMAT_SIGNED_INT32,
+    HIP_AD_FORMAT_HALF,          HIP_AD_FORMAT_FLOAT};
+
+// Helpful for printing errors
+const char* formatToString(hipArray_Format f) {
+  switch (f) {
```

Comment:
I'd recommend adding ```return "not found";``` as the default of the switch

Reply:
done.